Tools/scripts/configure-ci.sh

```diff
@@ -1,67 +1,14 @@
 #!/bin/bash
 # Install dependencies and configure the environment for CI build testing
 
-set -ex
-
-PKGS="build-essential gawk ccache genromfs libc6-i386 \
-      python-argparse python-empy python-serial python-pexpect python-dev python-pip zlib1g-dev gcc-4.9 g++-4.9"
-
-ARM_ROOT="gcc-arm-none-eabi-4_9-2015q3"
-ARM_TARBALL="$ARM_ROOT-20150921-linux.tar.bz2"
-
-RPI_ROOT="master"
-RPI_TARBALL="$RPI_ROOT.tar.gz"
-
-sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
-sudo apt-get -qq -y update
-sudo apt-get -qq -y install $PKGS
-sudo pip install mavproxy
-sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.9 90 \
-    --slave /usr/bin/g++ g++ /usr/bin/g++-4.9
-
-pushd $HOME
-
-mkdir -p $HOME/opt
-pushd $HOME/opt
-
-# PX4 toolchain
-compiler=$ARM_ROOT
-if [ ! -d "$HOME/opt/$compiler" ]; then
-  wget http://firmware.diydrones.com/Tools/PX4-tools/$ARM_TARBALL
-  tar -xf $ARM_TARBALL
-fi
-
-# RPi/BBB toolchain
-compiler="tools-master/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64"
-if [ ! -d "$HOME/opt/$compiler" ]; then
-  wget http://firmware.diydrones.com/Tools/Travis/NavIO/$RPI_TARBALL
-  tar -xf $RPI_TARBALL
+BASEDIR=$(dirname "$0")
+# Default to Linux on Semaphore builds
+if [ -z "$TRAVIS_OS_NAME" -o "$TRAVIS_OS_NAME" == "linux" ]; then
```

Comment:
The variable should be generic and don't depend on Travis. These scripts should be as much agnostic to the CI as possible, otherwise long term it is a pain to maintain. You could define an env var that will pass the OS name, just like we do for the build target.